lib/src/metta/text.rs

```diff
@@ -253,10 +258,12 @@ mod tests {
     }
 
     #[test]
-    #[should_panic(expected = "Unexpected right bracket")]
-    fn test_panic_on_unbalanced_brackets() {
-        let mut parser = SExprParser::new("(a))");
-        while let Some(_) = parser.parse(&Tokenizer::new()) {}
+    fn test_unbalanced_brackets() {
+        let mut parser = SExprParser::new("(a)))");
+        let x = parser.parse(&Tokenizer::new());
+        //println!("x: {:?}", x);
+
+        assert_eq!(Err(String::from("Unexpected right bracket")), parser.parse(&Tokenizer::new()));
```

Comment:
This test looks ok. I would reuse the instance of the tokenizer, remove the commented code and check the value of the first atom though:
```suggestion
    fn test_unbalanced_brackets() {
        let tokenizer = Tokenizer::new();
        let mut parser = SExprParser::new("(a)))");
        
        assert_eq!(parser.parse(&tokenizer), Ok(Some(expr!(("a")))));
        assert_eq!(parser.parse(&tokenizer), Err(String::from("Unexpected right bracket")));
```